extensions/SharkPool/BetterInput.js

```diff
@@ -0,0 +1,1305 @@
+          this.exitEffect = null;
+          break;
+        case "Fade out":
+          this.exitEffect = "fadeOut";
+          break;
+        case "Shrink":
+          this.exitEffect = "shrinkOut";
+          break;
+        case "Both":
+          this.exitEffect = "Both";
+          break;
+      }
+    }
+
+    setPrePosition(args) {
+      this.textBoxX = args.X;
+      this.textBoxY = args.Y * -1;
+    }
+
+    setPosition(args) {
+      this.textBoxX = args.X;
+      this.textBoxY = args.Y * -1;
+
+      const overlays = document.querySelectorAll(".ask-box");
+      overlays.forEach((overlay) => {
+        this.updateOverlayPosition(overlay);
+      });
+    }
+
+    changePosition(args) {
+      this.textBoxX = this.textBoxX + args.X;
+      this.textBoxY = this.textBoxY + args.Y * -1;
+
+      const overlays = document.querySelectorAll(".ask-box");
+      overlays.forEach((overlay) => {
+        this.updateOverlayPosition(overlay);
+      });
+    }
+
+    updateOverlayPosition(overlay) {
+      if (this.textBoxX !== null && this.textBoxY !== null) {
+        overlay.style.left = `${41 + this.textBoxX}%`;
+        overlay.style.top = `${44 + this.textBoxY}%`;
+      } else {
+        overlay.style.left = "50%";
+        overlay.style.top = "50%";
+      }
+    }
+
+    setMaxBoxCount(args) {
+      this.maxBoxCount = args.MAX;
+    }
+
+    setFontSize(args) {
+      this.fontSize = args.SIZE + "px";
+    }
+
+    setTextAlignment(args) {
+      this.textAlign = args.ALIGNMENT;
+    }
+
+    setFontFamily(args) {
+      this.fontFamily = args.FONT;
+    }
+
+    setSlider(args) {
+      this.minSlider = args.MIN;
+      this.maxSlider = args.MAX;
+      this.defaultSlider = args.DEFAULT;
+    }
+
+    setEnable(args) {
+      const enableMenu = args.ENABLE_MENU;
+      const action = args.ACTION;
+      switch (enableMenu) {
+        case "Button 2":
+          this.showCancelButton = action === "Enabled";
+          break;
+        case "Button 3":
+          this.showButton3 = action === "Enabled";
+          break;
+        case "Button 4":
+          this.showButton4 = action === "Enabled";
+          break;
+      }
+    }
+
+    setInputType(args) {
+      this.isInputEnabled = args.ACTION;
+    }
+
+    setButtonText(args) {
+      const buttonMenu = args.BUTTON_MENU;
+      const text = args.TEXT;
+      switch (buttonMenu) {
+        case "Button 1":
+          this.submitButtonText = text;
+          break;
+        case "Button 2":
+          this.cancelButtonText = text;
+          break;
+        case "Button 3":
+          this.Button3Text = text;
+          break;
+        case "Button 4":
+          this.Button4Text = text;
+          break;
+        case "Dropdown":
+          this.DropdownText = text;
+          break;
+      }
+    }
+
+    askAndWaitForInput(args) {
+      if (this.askBoxCount < this.maxBoxCount) {
+        return new Promise((resolve) => {
+          this.askAndWait(args).then(() => {
```

Comment:
pretty sure the whole new Promise thing isn't required since the thing you're wrapping is, itself, a promise

Reply:
I need it to be that way so it can properly handle the various inputs. That including the ask and wait reporter block... If Im wrong, what can I do?

Reply:
This was not resolved